components/feature/sitepermissions/src/test/java/mozilla/components/feature/sitepermissions/SitePermissionsDialogFragmentTest.kt

```diff
@@ -53,6 +53,30 @@ class SitePermissionsDialogFragmentTest {
         val checkBox = dialog.findViewById<CheckBox>(R.id.do_not_ask_again)
 
         assertTrue(checkBox.isVisible())
+        assertTrue(checkBox.isChecked)
+    }
+
+    @Test
+    fun `display dialog with remember choice unchecked`() {
+        val fragment = spy(
+                SitePermissionsDialogFragment.newInstance(
+                        "sessionId",
+                        "title",
+                        R.drawable.notification_icon_background,
+                        mock(),
+                        true,
+                        shouldSelectRememberChoice = false
+                )
+        )
+
+        doReturn(context).`when`(fragment).requireContext()
+        val dialog = fragment.onCreateDialog(null)
+        dialog.show()
+
+        val checkBox = dialog.findViewById<CheckBox>(R.id.do_not_ask_again)
+
+        assertTrue(checkBox.isVisible())
```

Comment:
I think we also have to check that `userSelectionCheckBox` gets initialize with the right value.

Reply:
Good point.